crypto/rand/rand_unix.c

```diff
@@ -325,6 +329,98 @@ static struct random_device {
 } random_devices[OSSL_NELEM(random_device_paths)];
 static int keep_random_devices_open = 1;
 
+#   if defined(DEVRANDOM_WAIT)
+static void *shm_addr;
+
+#    if !defined(FIPS_MODE)
+static void cleanup_shm(void)
+{
+    shmdt(shm_addr);
+}
+#    endif
+#   endif
```

Comment:
what came up from discussion with @t8m is this:
I'd like to move this endif to after the whole if defined(__linux) block:
```C
#   if defined(DEVRANDOM_WAIT)
static void *shm_addr;

#    if !defined(FIPS_MODE)
static void cleanup_shm(void)
{
    shmdt(shm_addr);
}
#    endif
...
#    if defined(__linux)
static int wait_random_seeded(void)
{
    static int seeded = OPENSSL_RAND_SEED_DEVRANDOM_SHM_ID < 0;
...
    return seeded;
}
#    else /* !defined(__linux) */
static int wait_random_seeded(void)
{
    return 1;
}
#   endif
#  else /* !defined(DEVRANDOM_WAIT) */
static int wait_random_seeded(void)
{
    return 1;
}
#  endif
```

Reply:
FYI the syntax I mentioned was swallowed in comment https://github.com/openssl/openssl/pull/9595#issuecomment-522601330:
`./config -DDEVRANDOM="\"\\\"/dev/urandom\\\"\""`

Reply:
I used the condition `defined(__linux)` instead of `defined(DEVRANDOM_WAIT)` which keeps things a bit simpler.

Reply:
ah, yes, that is probably better. However I would still add `&& defined(DEVRANDOM_WAIT)`
to the `#if`condition in order to keep supporting how the previous
patch was disabled, this meas now we will have two ways to disable
one being `./config -DDEVRANDOM_WAIT="\"\\\"/dev/urandom\\\"\""`
and `./config -DOPENSSL_RAND_SEED_DEVRANDOM_SHM_ID=-1`
(and later probably as @richsalz suggested a disable-devrandom-wait or something)

Reply:
Okay, I've pushed that change too -- `DEVRANDOM_WAIT` is conditioned upon `__linux` in the headers but it can't hurt.

Reply:
Thanks